app/models/group.rb

```diff
@@ -604,4 +604,12 @@ def limit_inheritance
   def self.with_one_coordinator
     published.select{ |g| g.admins.count == 1 }
   end
+
+  def default_cover_photo
+    if self.default_group_cover
+      /^.*(?=\?)/.match(self.default_group_cover.cover_photo.url).to_s
```

Comment:
What if they all have the same cover photo?.. ie: is there a database only migration I can help you write? `UPDATE groups set cover_photo_file_name = ? where cover_photo_file_name is null`
? Is it it more complex than that?